src/libraries/System.Reflection/tests/PointerTests.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Xunit;
+using System.Reflection;
+
+namespace System.Reflection.Tests
+{
+    public class PointerTests
+    {
+        public unsafe struct BitwiseComparable
+        {
+            public int* PublicInt;
+        }
+
+        public unsafe struct MemberwiseComparable
+        {
+            public string ReferenceType;
+            public int* PublicInt;
+        }
+
+        [Fact]
+        public unsafe void BitwiseEquality_AreEqual()
+        {
+            int someNumber = 1;
+            var a = new BitwiseComparable();
+            a.PublicInt = &someNumber;
+            var b = a;
+
+            Assert.True(a.Equals(b));
+        }
+
+        [Fact]
+        public unsafe void BitwiseEquality_EqualWithSelf()
+        {
+            int someNumber = 1;
+            var a = new BitwiseComparable();
+            a.PublicInt = &someNumber;
+
+            Assert.True(a.Equals(a));
+        }
+
+        [Fact]
+        public unsafe void MemberwiseEquality_AreEqual()
+        {
+            int someNumber = 1;
+            var a = new MemberwiseComparable();
+            a.PublicInt = &someNumber;
+            var b = a;
+
+            Assert.True(a.Equals(b));
+        }
+
+        [Fact]
+        public unsafe void MemberwiseEquality_EqualWithSelf()
+        {
+            int someNumber = 1;
+            var a = new MemberwiseComparable();
+            a.PublicInt = &someNumber;
+
+            Assert.True(a.Equals(a));
+        }
+
+        [Fact]
+        public unsafe void Nullptrs_AreEqual()
+        {
+            var a = Pointer.Box(null, typeof(int*));
+            var b = Pointer.Box(null, typeof(int*));
+
+            Assert.True(a.Equals(b));
+        }
+
```

Comment:
There should be a test that verifies the pointer type as mentioned above in the discussions.

For example, this test will pass but should it?
```cs
        [Fact]
        public unsafe void DifferentPointerTypes_Null_AreEqual()
        {
            var a = Pointer.Box(null, typeof(int*));
            var b = Pointer.Box(null, typeof(long*));

            Assert.True(a.Equals(b)); // Passes, but should it return false?
        }
```

Reply:
As I said above 

> I didn't add type checking because the unboxed pointers aren't type checked either

I could add type checking if you think that this is the better approach. It doesn't affect the bug fix.

Reply:
I don't think the type check is necessary, since treating a pointer like an "untyped" address should be sufficient and may be desired in some cases.

However, please add a test that asserts the behavior like the `DifferentPointerTypes_Null_AreEqual` I added above.